pkg/detectors/dockerhub/dockerhub_test.go

```diff
@@ -0,0 +1,137 @@
+//go:build detectors
+// +build detectors
+
+package dockerhub
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/kylelemons/godebug/pretty"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+func TestDockerhub_FromChunk(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+	testSecrets, err := common.GetSecret(ctx, "trufflehog-testing", "detectors2")
+	if err != nil {
+		t.Fatalf("could not get test secrets from GCP: %s", err)
+	}
+	username := testSecrets.MustGetField("DOCKERHUB_USERNAME")
+	email := testSecrets.MustGetField("DOCKERHUB_EMAIL")
+	password := testSecrets.MustGetField("DOCKERHUB_PASSWORD")
+	inactive_password := testSecrets.MustGetField("DOCKERHUB_INACTIVE_PASSWORD")
+
+	type args struct {
+		ctx    context.Context
+		data   []byte
+		verify bool
+	}
+	tests := []struct {
+		name    string
+		s       Scanner
+		args    args
+		want    []detectors.Result
+		wantErr bool
+	}{
+		{
+			name: "found, verified",
+			s:    Scanner{},
+			args: args{
+				ctx:    context.Background(),
+				data:   []byte(fmt.Sprintf("docker login -u %s -p %s", username, password)),
```

Comment:
question: Should we also include the PAT as part of the data here? It looks like the detector only checks for the PAT,  therefore we will need to also include the raw PAT in this []byte.

Ex:
```
pat := testSecrets.MustGetField("DOCKERHUB_PAT")
...
data:   []byte(fmt.Sprintf("docker %s login -u %s -p %s", pat, username, password)),
...
```

Reply:
The `DOCKERHUB_PASSWORD` field is meant to store the PAT (the PAT is used as a replacement for the password). Sorry for the misleading naming, changed in the commit!